aspnetcore/blazor/components/index.md

```diff
@@ -268,17 +268,141 @@ In the following example from the sample app, the `ParentComponent` sets the val
+
+  ```razor
+  <ChildComponent Title="@title">
+      Title from field.
+  </ChildComponent>
+  
+  @code {
+      private string title = "Panel Title from Parent";
+  }
+  ```
+
+* To assign the result of a parent component's method to a child component's parameter, prefix the method name with the `@` symbol. To assign the result of a C# method named `GetTitle` to the `Title` parameter, use the syntax `Title="@GetTitle"`, as seen in the following parent component:
+
+  ```razor
+  <ChildComponent Title="@GetTitle()">
+      Title from method.
+  </ChildComponent>
+  
+  @code {
+      private string GetTitle()
+      {
+          return "Panel Title from Parent";
+      }
+  }
+  ```
+  
+* To assign the result of an [implicit C# expression](xref:mvc/views/razor#implicit-razor-expressions) in the parent component to a child component's parameter, prefix the implicit expression with an `@` symbol. To assign the current local date in long format with <xref:System.DateTime.ToLongDateString%2A> in the parent component, assign `@DateTime.Now.ToLongDateString()` to the child's parameter. To assign a `person` object's `Name` property, assign `@person.Name`. The following example demonstrates these concepts:
+
+  ```razor
+  <ChildComponent Title="@DateTime.Now.ToLongDateString()">
+      Title from implicit Razor expression.
+  </ChildComponent>
+  
+  <ChildComponent Title="@person.Name">
+      Title from implicit Razor expression.
+  </ChildComponent>
+  ```
+  
+  Unlike in Razor pages (`.cshtml`), Blazor can't perform asynchronous work in a Razor expression while rendering a component. Instead, asynchronous work is performed during one of the [asynchronous component lifecycle events](xref:blazor/components/lifecycle). The following Razor syntax is **not** supported:
+  
+  ```razor
+  <ChildComponent Title="@await ...">
+      ...
+  </ChildComponent>
+  ```
+  
+  The code in the preceding example generates an IntelliSense error in Visual Studio and a *compiler error* if the app is built:
+  
+  > The 'await' operator can only be used within an async method. Consider marking this method with the 'async' modifier and changing its return type to 'Task'.
+
+  To obtain a value for the `Title` parameter in the preceding example asychronously, the component can use the [`OnInitializedAsync` lifecycle event](xref:blazor/components/lifecycle#component-initialization-methods), as the following example demonstrates:
+  
+  ```razor
+  <ChildComponent Title="@title">
+      Title from implicit Razor expression.
+  </ChildComponent>
+  
+  @code {
+      private string title;
+      
+      protected override async Task OnInitializedAsync()
+      {
+          title = await Task.FromResult<string>("Panel Title from Parent");
```

Comment:
The `Task.FromResult` snippet may be confusing because people might think they are supposed to use that in their code for some reason. It might be easier to follow like this:

```suggestion
          title = await ...;
```

... like you have in the Razor snippet above.